com.unity.sg2/Editor/GraphDeltaRegistry/FunctionDefinitions/StandardDefinitions/Input/Textures/SampleTriplanarNode.cs

```diff
@@ -0,0 +1,193 @@
+	int3 ma = (n.x>n.y && n.x>n.z) ? int3(0,1,2) : (n.y>n.z) ? int3(1,2,0) : int3(2,0,1);
+	int3 mi = (n.x<n.y && n.x<n.z) ? int3(0,1,2) : (n.y<n.z) ? int3(1,2,0) : int3(2,0,1);
+	int3 me = int3(3, 3, 3) - mi - ma;
+	//create coordinates
+	xCoords.x = Position[ma.y]; xCoords.y = Position[ma.z];
+	xDDX.x = dpdx[ma.y]; xDDX.y = dpdx[ma.z];
+	xDDY.x = dpdy[ma.y]; xDDY.y = dpdy[ma.z];
+	yCoords.x = Position[me.y]; yCoords.y = Position[me.z];
+	yDDX.x = dpdx[me.y]; yDDX.y = dpdx[me.z];
+	yDDY.x = dpdy[me.y]; yDDY.y = dpdy[me.z];	
+	// project+fetch
+	Node_X = SAMPLE_TEXTURE2D_GRAD(UnityBuildTexture2DStructNoScale(Texture).tex, Sampler.samplerstate, xCoords, xDDX, xDDY);
+	Node_Y = SAMPLE_TEXTURE2D_GRAD(UnityBuildTexture2DStructNoScale(Texture).tex, Sampler.samplerstate, yCoords, yDDX, yDDY);
+	// blend factors
+	w.x = n[ma.x]; 	w.y = n[me.x];
+	// make local support
+	w = saturate( (w-0.5773)/(1.0-0.5773));
+	// shape transition
+	w = pow( w, float2(contrast/8.0, contrast/8.0) );
+	// blend and return
+	XYZ = (Node_X*w.x + Node_Y*w.y) / (w.x + w.y);",
+                    new ParameterDescriptor("Texture", TYPE.Texture2D, Usage.In),
+                    new ParameterDescriptor("Sampler", TYPE.SamplerState, Usage.In),
+                    new ParameterDescriptor("Position", TYPE.Vec3, Usage.In, REF.WorldSpace_Position),
+                    new ParameterDescriptor("Normal", TYPE.Vec3, Usage.In, REF.WorldSpace_Normal),
+                    new ParameterDescriptor("Tile", TYPE.Float, Usage.In, new float[] { 1.0f }),
+                    new ParameterDescriptor("Contrast", TYPE.Float, Usage.In, new float[] { 1.0f }),
+                    new ParameterDescriptor("XYZ", TYPE.Vec4, Usage.Out),
+                    new ParameterDescriptor("n", TYPE.Vec3, Usage.Local),
+                    new ParameterDescriptor("dpdx", TYPE.Vec3, Usage.Local),
+                    new ParameterDescriptor("dpdy", TYPE.Vec3, Usage.Local),
+                    new ParameterDescriptor("Node_X", TYPE.Vec4, Usage.Local),
+                    new ParameterDescriptor("Node_Y", TYPE.Vec4, Usage.Local),
+                    new ParameterDescriptor("w", TYPE.Vec2, Usage.Local),
+                    new ParameterDescriptor("xCoords", TYPE.Vec2, Usage.Local),
+                    new ParameterDescriptor("xDDX", TYPE.Vec2, Usage.Local),
+                    new ParameterDescriptor("xDDY", TYPE.Vec2, Usage.Local),
+                    new ParameterDescriptor("yCoords", TYPE.Vec2, Usage.Local),
+                    new ParameterDescriptor("yDDX", TYPE.Vec2, Usage.Local),
+                    new ParameterDescriptor("yDDY", TYPE.Vec2, Usage.Local)
+                )
+            }
+        );
+
+        public static NodeUIDescriptor NodeUIDescriptor => new(
+            Version,
+            Name,
+            tooltip: "Samples a texture three times and projects in front/back, top/bottom, and left/right.",
+            categories: new string[2] { "Input", "Texture" },
```

Comment:
In sg1 it's under `UV` , do we want to move it to `Input`>`Texture`?

Reply:
I think we should move it, yes.  Do you? It doesn't create UV coordinates as output like the other UV nodes do.  It samples textures like the other Input/Texture nodes.  We're adding the UVTriplanar node to the UV category, so moving this one helps to differentiate them.

Reply:
sounds good!